### DISCUSSION TITLE
Unexpected negative and asymmetry variable values

### DISCUSSION THREAD

**Discussion by NorthMagic** [idaholab/moose#26078] [Q&A General]
Hi,

I'm trying to solve neutron diffusion equation by using MOOSE. Several kernels derived from `Kernel` were constructed for diffusion, collision, scattering and fission, and I verified the code by several eigenvalue problem benchmarks. Then I tried to expand it for transient problems. However, non-physical variable values (negative and asymmetry neutron fluxes) always occur.

To test the code, I designed a test case and tried to solve the neutron diffusion equation with `Steady` executioner in which a keff is fixed and comes from a reference solution. From my point, this equals to a time-step calculation in the transient problem. In the test case, the solution was converged but unexpected variable values were obtained. In a strange way, the correct solutions could be obtained by `Eigenvalue` executioner. The difference between the mentioned two calculation is mainly the implementation of `Kernel` for the fission source term in the neutron diffusion equation. I checked it carefully and no mistakes were found in the fission kernel. 

Could you give me some advice? Thanks for your help.

**Comment by GiudGiud** [comment]
Hello

Using a Steady executioner would be an infinite time step. I would not expect a good value to be output. Depending on the keff and the rounding floating point error on how it is input to normalize the fission source, the solution would either be 0 or infinite. 

Now when using a time dependent solve:
- did you load the eigenvalue solve solution ? Did you normalize the fission source?
- how did you implement the time derivative term?
- what kind of time integration scheme are you using?

Guillaume
